src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java

```diff
@@ -469,10 +469,25 @@ private boolean inACycle2(boolean includeMetaData, Wire wire, Bytes<?> bytes) th
         return true;
     }
 
+    private AcknowledgedIndexReplicatedCheck acknowledgedIndexReplicatedCheck
+            = (index, lastSequenceAck) -> index == lastSequenceAck;
+
     private boolean inACycleCheckRep() {
         final long lastSequenceAck = queue.lastAcknowledgedIndexReplicated();
         final long index = index();
-        return index > lastSequenceAck;
+        if (lastSequenceAck == -1)
+            return true; // don't allow the message to be read
+        if (index == Long.MIN_VALUE)
+            return false; // allow the message to be read
+
+        // check both are in the same roll cycle
+        if (queue.rollCycle().toCycle(index) != queue.rollCycle().toCycle(lastSequenceAck))
```

Comment:
If the implementation doesn't support calculating the delta across roll cycles I reckon we should call that out on the javadoc (if not already done)

Reply:
Agreed